server/crawlers/redditCrawler.js

```diff
@@ -0,0 +1,40 @@
+const snoowrap = require("snoowrap");
+
+async function redditCrawler(companyName, numOfMentionsToObtain) {
+  const scraper = new snoowrap({
+    userAgent: process.env.Reddit_userAgent,
+    clientId: process.env.Reddit_clientId,
+    clientSecret: process.env.Reddit_clientSecret,
+    refreshToken: process.env.Reddit_refreshToken,
+  });
+
+  const subreddit = await scraper.getSubreddit("all");
+  const topPosts = await subreddit.getTop({
+    time: "all",
+    limit: numOfMentionsToObtain,
+  });
+
+  let data = [];
```

Comment:
That only gets the top 20ish posts though, no? Instead we want want to use the /search endpoint

Reply:
When I tested it, `getTop` returned as many as 954 mentions, I thought it would be sufficient. But now I have reconsidered it and realized it may be not. I just tested using `getHot` requiring 5000 mentions, and the list of results was just 5000 long. So I guess we can use `getHot` instead?

Reply:
Is there a way to perform an actual search by name within a subreddit? Reddit has an api endpoint for that

Reply:
I have added a command `npm run scraper`  to test the scraper, which will require 1000 mentions and print them(only those with content) onto the console. I am not sure familiar with node.js, not sure if this is the best way to run it.

Reply:
Have changed from `getHot` to `search`.
The results will be sorted by relevance, and each mention's title must contain the specified company name.
Issuing `npm run scraper` will do the search based on "Twitter" and print the results on the console.